mobile/src/bundles/common/components/searchable-dropdown/searchable-dropdown.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+import {
+    type Control,
+    type FieldPath,
+    type FieldValues,
+} from 'react-hook-form';
+import { TextInput } from 'react-native';
+
+import {
+    ScrollView,
+    Text,
+    TouchableOpacity,
+    View,
+} from '~/bundles/common/components/components';
+import { TextCategory } from '~/bundles/common/enums/enums';
+import { useFormController } from '~/bundles/common/hooks/hooks';
+import { globalStyles } from '~/bundles/common/styles/styles';
+
+import { styles } from './styles';
+
+type Properties<T extends FieldValues> = {
+    control: Control<T, null>;
+    name: FieldPath<T>;
+    hasError?: boolean;
+    items: string[];
+    placeholder?: string;
+    onItemSelect: (item: string) => void;
+};
+
+const SearchableDropdown = <T extends FieldValues>({
+    control,
+    name,
+    hasError,
+    items,
+    placeholder,
+    onItemSelect,
+}: Properties<T>): JSX.Element => {
+    const { field } = useFormController({ name, control });
+    const { value, onChange, onBlur } = field;
+    const [selectedItems, setSelectedItems] = useState<string[]>([]);
+    const [isListVisible, setIsListVisible] = useState<boolean>(false);
+
+    const toggleIsListVisible = (): void => {
+        setIsListVisible((previous) => !previous);
+    };
+
+    const handleItemSelect = (item: string): void => {
+        setSelectedItems([...selectedItems, item]);
+        onItemSelect(item);
+        toggleIsListVisible();
+    };
+
+    const filterItems = (items: string[]): string[] => {
+        if (typeof value !== 'string') {
+            return items;
+        }
+        return items.filter((item) =>
+            item.toLowerCase().includes(value.toLowerCase()),
+        );
+    };
+
+    return (
+        <View style={styles.container}>
+            <TextInput
+                placeholder={placeholder}
+                onChangeText={onChange}
```

Comment:
I don't understand what you mean

Reply:
If you have for example ["value1", "value2"] in form values and then you start typing to search for another value, your form value will be replaced on "searched value" 
So you have to remove onChangeText={onChange} to avoid such behavior. Try console log value for this dropdown and you'll see